fsspec/implementations/tests/test_http.py

```diff
@@ -280,6 +280,16 @@ def test_content_length_zero(server):
         assert f.read() == data
 
 
+def test_content_encoding_gzip(server):
+    h = fsspec.filesystem(
+        "http", headers={"give_length": "true", "gzip_encoding": "true"}
+    )
+    url = server + "/index/realfile"
+
+    with h.open(url, "rb") as f:
```

Comment:
`f` is a streaming type file here, with `None` for the size? Should probably assert that.

Reply:
Ah, not super familiar with fsspec, yet - Only tried to mimic the test above this one. Added assertions for type and size, now.